roachtest: avoid large payloads when testing long revision histories

renatolabs opened the pull request:
This commit updates the `backup-restore/mixed-version` roachtest to avoid using large payloads (128KiB) in the bank workload when also testing long revision histories (100 rows). This often leads to failures such as `split failed... cannot find valid split key`.

Informs: #102284

Release note: None

Commit:
roachtest: avoid large payloads when testing long revision histories

This commit updates the `backup-restore/mixed-version` roachtest to
avoid using large payloads (128KiB) in the bank workload when also
testing long revision histories (100 rows). This often leads to
failures such as `split failed... cannot find valid split key`.

Informs: #102284

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106539)
<!-- Reviewable:end -->

msbutler commented:
we have a duplicate tracking issue https://github.com/cockroachdb/cockroach/issues/102284

renatolabs commented:
> we have a duplicate tracking issue #102284

Thanks, I had forgotten about that issue. Updated the commit to mention that instead, and also added a `TODO` to revert this change if that is ever addressed.

renatolabs commented:
blathers backport 23.1
